app/infrastructure/database/repositories.py

```
@@ -0,0 +1,265 @@
+            repo = await Repo.filter(repo_id=repo_id).first()
+            if not repo:
+                raise RepoNotFoundError(f"Repository {repo_id} not found")
+
+            repo.status = status
+            for key, value in kwargs.items():
+                if hasattr(repo, key):
+                    setattr(repo, key, value)
+
+            await repo.save()
+            logger.info(f"Updated repo {repo_id} status to {status}")
+        except RepoNotFoundError:
```

Ruff (0.11.9)

`RepoNotFoundError` may be undefined, or defined from star imports

(F405)